clang/lib/Sema/SemaLambda.cpp

```diff
@@ -1246,7 +1246,12 @@ void Sema::ActOnLambdaExpressionAfterIntroducer(LambdaIntroducer &Intro,
 
       if (auto *BD = R.getAsSingle<BindingDecl>())
         Var = BD;
-      else
+      else if (auto *FD = R.getAsSingle<FieldDecl>()) {
+        Var = R.getAsSingle<VarDecl>();
```

Comment:
Don’t think there is a reason to do this since we’ll be `continue`’ing below anyway.

Reply:
To clarify since Github’s formatting is a bit jank here, I’m talking about the `Var = R.getAsSingle<VarDecl>();` part

Reply:
Could you explain what kinds of tests or what to verify? I tested the scenario in the issue.

Reply:
> Could you explain what kinds of tests or what to verify? I tested the scenario in the issue.

Well, we need to add tests that verify that this actually solves the issue (i.e. that we emit the diagnostic if someone tries to capture a class member); you can take a look at some of the files in `clang/test/SemaCXX` to see how that is usually done.